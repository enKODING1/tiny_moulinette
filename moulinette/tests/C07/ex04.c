#include <stdio.h>

char *ft_convert_base(char *nbr, char *base_from, char *base_to);

void test(char *nbr, char *base_from, char *base_to)
{
    char *res;
    res = ft_convert_base(nbr, base_from, base_to);
    if (res == NULL)
    {
        printf("NULL\n");
        return;
    }
    printf("%s\n", res);
}

int main(void)
{
    test("4242", "0123456789", "");
    test("4242", "0123456789", "01");
    test("4242", "0123456789", "0+");
    test("4242", "0123456789", "0-");
    test("4242", "0123456789", "0 ");
    test("4242", "0123456789", "0\t");
    test("4242", "0123456789", "0\n");
    test("4242", "0123456789", "0\v");
    test("4242", "0123456789", "0\f");
    printf("---\n");
    test("1000010010010", "", "0123456789");
    test("1000010010010", "01", "0123456789");
    test("1000010010010", "0+", "0123456789");
    test("1000010010010", "0-", "0123456789");
    test("1000010010010", "0 ", "0123456789");
    test("1000010010010", "0\t", "0123456789");
    test("1000010010010", "0\n", "0123456789");
    test("1000010010010", "0\v", "0123456789");
    test("1000010010010", "0\f", "0123456789");
    printf("---\n");
    test("4242", "abcdefghij", "abcdefghij");
    printf("---\n");
    test(" \t\n\v\f+1234", "0123456789", "0123456789");
    test(" \t\n\v\f+-1234", "0123456789", "0123456789");
    test(" \t\n\v\f+--1234", "0123456789", "0123456789");
    test(" \t\n\v\f-+-1234", "0123456789", "0123456789");
    test(" \t\n\v\f-+--1234", "0123456789", "0123456789");
    test(" \t\n\v\f-+-1234a", "0123456789", "0123456789");
    test(" \t\n\v\f-+-123a4", "0123456789", "0123456789");
    test(" \t\n\v\f-+-12a34", "0123456789", "0123456789");
    test(" \t\n\v\f-+-1a234", "0123456789", "0123456789");
    test(" \t\n\v\f-1234-", "0123456789", "0123456789");
    test(" \t\n\v\f+1234-", "0123456789", "0123456789");
    printf("---\n");
    test(" \t\n\v\f-+-IOI", "OI", "0123456789");
    test(" \t\n\v\f-+-!@!", "@!", "0123456789");
    test(" \t\n\v\f-+-^%^", "%^", "0123456789");
    printf("---\n");
    return 0;
}
