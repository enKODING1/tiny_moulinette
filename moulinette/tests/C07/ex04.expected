NULL
1000010010010
NULL
NULL
NULL
NULL
NULL
NULL
NULL
---
NULL
0
NULL
NULL
NULL
NULL
NULL
NULL
NULL
---
a
---
1234
-1234
1234
1234
-1234
1234
123
12
1
-1234
1234
---
5
5
5
---
